### PR TITLE
BREAKING CHANGE(web-react): Rename `DropdownModern` to `Dropdown`

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

- `UncontrolledDropdown` was already in new syntax.
-  Adjustments were needed in `DropdownTrigger` and `DropdownPopover` to make styleprops and restprops work, plus added tests.
- Changes in Storybook.

### Additional context

TODO: 

- [x] delete and rename `DropdownModern`
- [x] Create codemod for renaming DropdownModer -> Dropdown
- [x] UncontrolledDropdown
- [x] Migration guide

### Issue reference

[Nahradit DropdownModern -> Dropdown](https://jira.almacareer.tech/browse/DS-1192)

### CODE REVIEW

File: packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx [hunk 1]
```diff
@@ -5,57 +5,64 @@ import { classNamePrefixProviderTest } from '../../../../tests/providerTests/cla
...
   stylePropsTest(
-    (props: Record<string, unknown>) => <Dropdown {...props} data-testid="test-dropdown" />,
+    (props: Record<string, unknown>) => (
+      <Dropdown id="dropdown" isOpen={false} onToggle={() => {}} {...props} data-testid="test-dropdown" />
+    ),
     'test-dropdown',
   );
 
-  restPropsTest(Dropdown, '.Dropdown');
+  restPropsTest(Dropdown, '.DropdownWrapper');
```
Comment (literat) [inline packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx:21]: Is there still a `DropdownWrapper` class?

**Comment by literat** [comment]
Please, use English for PR description :-)

**Comment by literat** [comment]
To the commit message: it is not just renaming, we are replacing an entire legacy dropdown with the new one.

**Comment by literat** [comment]
However, everything else is working and it is looking good for me. I will give this a second try yesterday. Good Job 👍

**Comment by pavelklibani** [comment]
> Please, use English for PR description :-)

Changed to english

> To the commit message: it is not just renaming, we are replacing an entire legacy dropdown with the new one.

Commit message changed

**Comment by literat** [comment]
Can you please remove the `defaultProps` in the `DropdownTrigger` in favor fo default values?

File: packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx (hunk 1 above)
Comment (literat) [reply packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx:21]: Maybe we should narrow this down in the future. What do you think @adamkudrna @crishpeen ?

**Comment by literat** [comment]
Just wonder based on Git and how GitHub displays it, how was your approach to this adjustment?
Remove `Dropdown` and then rename `DropdownModern` to `Dropdown`? Because based on the Git it looks like the `DropdownModern` was deleted and changes were applied to `Dropdown` so that is why I am interested. Trying to understand how Git is working with this kind of change. :-)

**Review by literat — APPROVED** [review]
LGTM 👍

File: packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx (hunk 1 above)
Comment (adamkudrna) [reply packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx:21]: > yes we have `DropdownWrapper` for `<Dropdown />` and `Dropdown` for `<DropdownPopover />`

Sounds like we should fix the naming 🙂. In CSS but also in Figma.

File: packages/web-react/src/components/Dropdown/demo/DropdownPlacements.tsx [hunk 2]
```diff
@@ -1,83 +1,81 @@
...
-            isChecked={placement === 'top-start'}
+            isChecked={placement === 'top-left'}
             isLabelHidden
             onChange={handlePlacementChange}
-            id="placement-top-start"
-            label="top-start"
-            value="top-start"
+            id="placement_top_left"
+            label="top-left"
+            value="top-left"
```
Comment (adamkudrna) [inline packages/web-react/src/components/Dropdown/demo/DropdownPlacements.tsx:31]: ↕️ All these placements should be flow-relative, i.e. they were correct before these changes happened.

```suggestion
            id="placement-top-start"
            label="top-start"
            value="top-start"
```

File: packages/web-react/src/components/Dropdown/demo/DropdownEnhancedShadow.tsx
Comment (adamkudrna) [inline packages/web-react/src/components/Dropdown/demo/DropdownEnhancedShadow.tsx:1]: I think this BC has been already merged: #1349

**Review by adamkudrna — APPROVED** [review]
Please fix the placement names, everything else looks OK. 👍🏻

File: packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          },
+        })
+        .forEach((openingTagPath) => {
+          // Change component name to 'Dropdown'
+          if (openingTagPath.node.name.type === 'JSXIdentifier') {
+            (openingTagPath.node.name as JSXIdentifier).name = 'Dropdown';
+          }
+        });
+
+      // Find closing tags for Tooltip components
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts]: changed

File: packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    const componentSpecifier = importStatements.find(j.ImportSpecifier, {
+      imported: {
+        type: 'Identifier',
+        name: 'DropdownModern',
+      },
+    });
+
+    // Check if Button specifier is present
+    if (componentSpecifier.length > 0) {
+      // Find opening tags for Tooltip components
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts]: changed

File: packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  // Check if the module is imported
+  if (importStatements.length > 0) {
+    const componentSpecifier = importStatements.find(j.ImportSpecifier, {
+      imported: {
+        type: 'Identifier',
+        name: 'DropdownModern',
+      },
+    });
+
+    // Check if Button specifier is present
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/codemods/src/transforms/v2/web-react/dropdownmodern-component-name.ts]: changed

File: packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Dropdown/__tests__/Dropdown.test.tsx:21]: I will create an issue for Thursday's lengthy themes.

**Pull request merged**